martin/src/bin/martin.rs

```diff
@@ -35,20 +34,15 @@ async fn start(args: Args) -> MartinResult<Server> {
     let (server, listen_addresses) = new_server(config.srv, sources)?;
     info!("Martin has been started on {listen_addresses}.");
     info!("Use http://{listen_addresses}/catalog to get the list of available sources.");
-
-    Ok(server)
+    server.await
 }
 
 #[actix_web::main]
 async fn main() {
     let env = env_logger::Env::default().default_filter_or("martin=info");
     env_logger::Builder::from_env(env).init();
 
-    start(Args::parse())
-        .await
-        .unwrap_or_else(|e| on_error(e))
-        .await
-        .unwrap_or_else(|e| on_error(e));
+    start(Args::parse()).await.unwrap_or_else(|e| on_error(e));
```

Comment:
probably better to inline the on_error here. Also, I noticed the same issue in `martin-cp` if you want to inline it there at the same time.

```suggestion
    start(Args::parse()).await.unwrap_or_else(|e| {
        if log_enabled!(log::Level::Error) {
            error!("{e}");
        } else {
            eprintln!("{e}");
        }
        std::process::exit(1);
    });
```